client/src/components/Reports/ReportsContainer.js

```diff
@@ -1,108 +1,94 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import "./ReportsContainer.css";
 import { isAdminUser } from "../../utils/AdminReportingRoles";
 import { FromReports } from "./FormReports";
 import { TrainingReports } from "./TrainingReports";
 
-class ReportsContainer extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showForms: false,
-      showTrainings: false,
-    };
-  }
+const ReportsContainer = (props) => {
+  const [{ showForms, showTrainings }, setState] = useState({
+    showForms: false,
+    showTrainings: false,
+  });
 
-  showFormReports = () => {
-    this.setState({ ...this.state, showForms: true, showTrainings: false });
+  const showFormReports = () => {
+    setState({ showForms: true, showTrainings: false });
   };
-  showTrainingReports = () => {
-    this.setState({ ...this.state, showForms: false, showTrainings: true });
+  const showTrainingReports = () => {
+    setState({ showForms: false, showTrainings: true });
   };
-  resetReports = () => {
-    this.setState({ ...this.state, showForms: false, showTrainings: false });
+  const resetReports = () => {
+    setState({ showForms: false, showTrainings: false });
   };
 
-  render() {
-    if (!this.state.showForms && !this.state.showTrainings) {
-      return (
-        <div style={{ marginTop: "50px" }}>
-          <div className="row" style={{ margin: "0px 30px" }}>
-            <div className="formTitleDiv" style={{ width: "100%" }}>
-              <h2 className="formTitle">Reports{"  "}</h2>
-              <hr />
-              <h2 className="formTitle">
-                <div
-                  style={{
-                    width: "100%",
-                    display: "flex",
-                    justifyContent: "space-evenly",
+  return !showForms && !showTrainings ? (
```

Comment:
The logic makes sense to me but I believe the render method could be simplified. Maybe call a function that has an if-else statement  along those lines

Reply:
I componentized the ReportsContainer component by creating an AllReports Container that contains both buttons to show Internal Trainings and Form Reports. I kept the hook and state in the ReportsContainer (parent) and passed down the state to the child (AllReports). This makes the ternary logic in the ReportsContainer component a lot easier to read.